evm_loader/program/src/evm/buffer.rs

```diff
@@ -48,13 +48,17 @@ impl Buffer {
         Buffer { ptr, len, inner }
     }
 
+    /// # Safety
+    ///
+    /// This function was marked as unsafe until correct lifetimes will be set.
+    /// At the moment, `Buffer` may outlive `account`, since no lifetimes has been set,
+    /// so they are not checked by the compiler and it's the user's responsibility to take
+    /// care of them.
     #[must_use]
-    pub fn from_account(account: &AccountInfo, range: Range<usize>) -> Self {
-        let data = unsafe {
-            // todo cell_leak #69099
-            let ptr = account.data.as_ptr();
-            (*ptr).as_mut_ptr()
-        };
+    pub unsafe fn from_account(account: &AccountInfo, range: Range<usize>) -> Self {
```

Comment:
Here it's not clear to me if it is safer to let the unsafe block inside the function or to make the whole function unsafe.

When you say it's the user's responsibility to take care of them, how does the code currently not use dangling pointers by accident? Where should I pay special attention?

I am asking this because the code around these unsafe blocks is unfamiliar to me.

Reply:
It's safer to have safe function which is really safe. But if we want to make it safe, we need to introduce lifetime for `Buffer` and borrow account for the whole lifetime of `Buffer`, otherwise, account may be destroyed or moved during the lifetime of `Buffer` and we can get dangling pointer. It might be possible to make this changes, but very labor intensive to implement. We shouldn't lie to the user that this function is safe and should at least attention them to use it with care.

Reply:
Ok. I got it. So we are safer with an `unsafe` function.